cmd/upload.go

```diff
@@ -113,19 +113,163 @@ var uploadCmd = &cobra.Command{
-		sync, _ := cmd.Flags().GetBool("sync")
 		chunkSize, _ := cmd.Flags().GetInt("chunksize")
 
-		if live {
-			// capture video and audio from local default camera and micrlphone, and upload it to zcn
-			err = startLiveUpload(cmd, allocationObj, localpath, remotepath, encrypt, chunkSize, attrs)
-		} else if sync {
-			// download video from remote live feed(eg youtube), and sync it to zcn
-			err = startSyncUpload(cmd, allocationObj, localpath, remotepath, encrypt, chunkSize, attrs)
-		} else {
-			err = startChunkedUpload(cmd, allocationObj, localpath, thumbnailpath, remotepath, encrypt, chunkSize, attrs, statusBar, false)
+		err = startChunkedUpload(cmd, allocationObj, localpath, thumbnailpath, remotepath, encrypt, chunkSize, attrs, statusBar, false)
+
+		if err != nil {
+			PrintError("Upload failed.", err)
+			os.Exit(1)
+		}
+		wg.Wait()
+		if !statusBar.success {
+			os.Exit(1)
+		}
+
+		if commit {
+			remotepath = zboxutil.GetFullRemotePath(localpath, remotepath)
+			statusBar.wg.Add(1)
+			commitMetaTxn(remotepath, "Upload", "", "", allocationObj, nil, statusBar)
+			statusBar.wg.Wait()
+		}
+	},
+}
+
+// feedCmd represents upload command with --sync flag
+var feedCmd = &cobra.Command{
+	Use:   "feed",
+	Short: "download segment files from remote live feed, and upload",
+	Long:  "download segment files from remote live feed, and upload",
+	Args:  cobra.MinimumNArgs(0),
+	Run: func(cmd *cobra.Command, args []string) {
+		fflags := cmd.Flags()              // fflags is a *flag.FlagSet
+		if !fflags.Changed("allocation") { // check if the flag "path" is set
+			PrintError("Error: allocation flag is missing") // If not, we'll let the user know
+			os.Exit(1)                                      // and return
+		}
+		if !fflags.Changed("remotepath") {
+			PrintError("Error: remotepath flag is missing")
+			os.Exit(1)
+		}
+
+		if !fflags.Changed("localpath") {
+			PrintError("Error: localpath flag is missing")
+			os.Exit(1)
+		}
+
+		allocationID := cmd.Flag("allocation").Value.String()
+		allocationObj, err := sdk.GetAllocation(allocationID)
+		if err != nil {
+			PrintError("Error fetching the allocation.", err)
+			os.Exit(1)
+		}
+		remotepath := cmd.Flag("remotepath").Value.String()
+		localpath := cmd.Flag("localpath").Value.String()
+		encrypt, _ := cmd.Flags().GetBool("encrypt")
+		commit, _ := cmd.Flags().GetBool("commit")
+
+		wg := &sync.WaitGroup{}
+		statusBar := &StatusBar{wg: wg}
+		wg.Add(1)
+		if strings.HasPrefix(remotepath, "/Encrypted") {
+			encrypt = true
+		}
+		var attrs fileref.Attributes
+		if fflags.Changed("attr-who-pays-for-reads") {
+			var (
+				wp  common.WhoPays
+				wps string
+			)
+			if wps, err = fflags.GetString("attr-who-pays-for-reads"); err != nil {
+				log.Fatalf("getting 'attr-who-pays-for-reads' flag: %v", err)
+			}
+			if err = wp.Parse(wps); err != nil {
+				log.Fatal(err)
+			}
+			attrs.WhoPaysForReads = wp // set given value
+		}
+
+		chunkSize, _ := cmd.Flags().GetInt("chunksize")
```

Comment:
```suggestion
		chunkSize, err := cmd.Flags().GetInt("chunksize")
		if err != nil {
		  log.Debug(err)
		}
```

More information are always better